### PR TITLE
Add the possiblity to pass a onChange callback

### PR DESCRIPTION
**Pull request by sventschui**
This will allow to integrate this component with redux-form.com

### CODE REVIEW

**Comment by tvararu** [comment]
Hey, isn't the `onConfirm` option the same as this? https://github.com/alphagov/accessible-autocomplete#analytics-and-tracking

**Comment by sventschui** [comment]
Not really. The big difference is that onConfirm is fired whenever an option is selected. I actually need an event when a character is entered (keyboard, speech, selecting an option, etc.)

**Comment by tvararu** [comment]
Makes sense! Sorry for the stupid question. 😄

File: src/autocomplete.js [hunk 1]
```diff
@@ -271,6 +275,10 @@ export default class Autocomplete extends Component {
       query: newQuery,
       selected: -1
     })
+
+    if (this.props.onChange) {
```
Comment (tvararu) [inline src/autocomplete.js]: You can avoid this check by defining `onChange` in `defaultProps` as `() => {}`.

This is consistent with `onConfirm` so I think we should do this.

Also will need a README update.

**Comment by sventschui** [comment]
@tvararu Finally found the time to update the PR

**Comment by sventschui** [comment]
@tvararu can we merge this one?

**Comment by danmitchell-** [comment]
Sorry for the delay @sventschui. Theo isn't at GDS anymore so we're working through the active PR's 👍

**Comment by danmitchell-** [comment]
@sventschui can you rebase this with master and i'll merge it, thanks. Looking to do a new release so we might as well include this 👍

**Comment by andreaswilli** [comment]
@dankmitchell can we merge this?

**Comment by 36degrees** [comment]
Oliver Byford (Ollie) [3:26 PM]
Hi,

In order to get this merged I think we’d need:

- this branch to be rebased against master and any conflicts resolved
- tests added that cover the new functionality introduced in this PR
- if possible, the commit history cleaned up to remove the merge from master

Unfortunately our team don’t have the time at the minute to be able to help with this. Would anyone be interested in picking this up so we can push it forward? @sventschui or @andreaswilli perhaps?

I’m conscious this PR is quite old and hasn’t had any activity for a while – we’ll keep it open for another couple of weeks, and if there’s been no activity we’ll close it and turn it into an issue.

Thanks.

**Comment by robinwhittleton** [comment]
We effectively reimplemented this with our use of this component, but I doubt our changes could be lifted into a solid PR.

If I get a chance I’ll rewrite what we did and submit it as a separate PR.

**Comment by nickcolley** [comment]
Going to close this out as there's not been activity, but I've referenced this in the feature request issue for event hooks: https://github.com/alphagov/accessible-autocomplete/issues/117

**Pull request closed without merging**